### PR TITLE
TE 5830 NewRelic monitoring

### PR DESCRIPTION
**Pull request by stereomon**
- Developer(s): @stereomon

- Ticket: https://spryker.atlassian.net/browse/TE-5830

- Academy PR: ACADEMY_URL_HERE

- Release Group: URL_HERE

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   NewRelic              | patch                 |                       |

-----------------------------------------

##### Change log

Fixes

- Fixed dependencies.

### CODE REVIEW

File: src/SprykerEco/Zed/NewRelic/Business/Model/RecordDeployment.php [hunk 1]
```diff
@@ -13,8 +13,8 @@
 
 class RecordDeployment implements RecordDeploymentInterface
 {
-    const STATUS_CODE_SUCCESS = 200;
-    const STATUS_CODE_REDIRECTION = 300;
+    protected const STATUS_CODE_SUCCESS = 200;
+    protected const STATUS_CODE_REDIRECTION = 300;
```
Comment (demkos) [inline src/SprykerEco/Zed/NewRelic/Business/Model/RecordDeployment.php]: Should it be public for BC reasons?

Comment (stereomon) [reply src/SprykerEco/Zed/NewRelic/Business/Model/RecordDeployment.php]: IMO not as this is for internal use only and the code is not used by any project ATM as it was not working.

File: src/SprykerEco/Zed/NewRelic/Business/Model/RecordDeployment.php [hunk 2]
```diff
@@ -55,9 +54,7 @@ public function __construct(
     /**
      * @param array $arguments
      *
-     * @throws \SprykerEco\Zed\NewRelic\Business\Exception\RecordDeploymentException
-     *
-     * @return $this
+     * @return \SprykerEco\Zed\NewRelic\Business\Model\RecordDeploymentInterface
```
Comment (dereuromark) [inline src/SprykerEco/Zed/NewRelic/Business/Model/RecordDeployment.php]: That looks wrong. This does return $this.
The return typehint should be removed instead.

File: src/SprykerEco/Zed/NewRelic/Communication/Console/RecordDeploymentConsole.php [hunk 3]
```diff
@@ -17,23 +17,23 @@
...
 
-    const ARGUMENT_DESCRIPTION = 'description';
-    const ARGUMENT_DESCRIPTION_DESCRIPTION = 'Deployment description';
+    protected const ARGUMENT_DESCRIPTION = 'description';
+    protected const ARGUMENT_DESCRIPTION_DESCRIPTION = 'Deployment description';
 
-    const ARGUMENT_CHANGELOG = 'changelog';
-    const ARGUMENT_CHANGELOG_DESCRIPTION = 'Change log';
+    protected const ARGUMENT_CHANGELOG = 'changelog';
+    protected const ARGUMENT_CHANGELOG_DESCRIPTION = 'Change log';
```
Comment (demkos) [inline src/SprykerEco/Zed/NewRelic/Communication/Console/RecordDeploymentConsole.php:36]: These consts also need to be fixed.

Comment (dereuromark) [reply src/SprykerEco/Zed/NewRelic/Communication/Console/RecordDeploymentConsole.php:36]: I think here it is fair to say those should be internal. They don't leave the scope here for public usage.

Comment (dereuromark) [reply src/SprykerEco/Zed/NewRelic/Communication/Console/RecordDeploymentConsole.php:36]: But the word here is wrong. Changelog is one word.

**Review by demkos — APPROVED** [review]

**Pull request merged**